scripts/generate-doc.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Shell script to generate the Terraform documentation for the resource and data sources.
+#
+# Usage: ./generate-doc.sh" ${resource_name}
+#   resource_name is the terraform resource name. Example: search_deployment
+#   echo "Examples:"
+#   echo "  search_deployment"
+#   echo "  project"
+#   echo "  online_archive"
+#   echo "  encryption_at_rest"
+#
+# The scripts requires to install tfplugindocs and to create the resource templates in 
+# templates/resources/${resource_name}.html.markdown.tmpl and 
+# templates/data-sources/${resource_name}.html.markdown.tmpl
+# templates/data-sources/${resource_name}s.html.markdown.tmpl
+
+set -euo pipefail
+
+TF_VERSION="${TF_VERSION:-"1.6.6"}" # TF version to use when running tfplugindocs. Default: 1.6.6
```

Comment:
i think it'll be hard to remember to update this version but I understand we only have TF_VERSION_LATEST in GH not in local / makefile.   

Can you try to see if it works with 1.6.x (for latest patch in 1.6) so we won't need to change it so often?

Reply:
Curious if the terraform version has much impact in generated docs. Having a fixed version does help ensure consistent results.

Reply:
that's a good point, i have no idea. on the hand there can be interesting improvements that we miss if you don't try with latest versions

Reply:
> Curious if the terraform version has much impact in generated docs. Having a fixed version does help ensure consistent results.

Unfortunately, their documentation does not explain how TF is used for the generation. I created https://github.com/hashicorp/terraform-plugin-docs/issues/323 to get an answer from HashiCorp

Reply:
> Can you try to see if it works with 1.6.x (for latest patch in 1.6) so we won't need to change it so often?

it did work indeed. Thanks! I will update the code

```
exporting schema from Terraform
compiling provider "mongodbatlas"
downloading Terraform CLI binary version from releases.hashicorp.com: 1.6
```

Reply:
Updated